packages/crayons-core/src/components/country-phone/country-phone.tsx

```diff
@@ -0,0 +1,339 @@
+   */
+  @Prop() readonly = false;
+
+  /**
+   * Specifies the input box as a mandatory field and displays an asterisk next to the label. If the attribute’s value is undefined, the value is set to false.
+   */
+  @Prop() required = false;
+
+  /**
+   * Specifies the input box as a mandatory field and displays an asterisk next to the label. If the attribute’s value is undefined, the value is set to false.
+   */
+  @Prop() requiredInnerHint = false;
+
+  /**
+   * Disables the component on the interface. If the attribute’s value is undefined, the value is set to false.
+   */
+  @Prop() disabled = false;
+
+  /**
+   * Hint text displayed below the text box.
+   */
+  @Prop() hintText = '';
+  /**
+   * Warning text displayed below the text box.
+   */
+  @Prop() warningText = '';
+  /**
+   * Error text displayed below the text box.
+   */
+  @Prop() errorText = '';
+
+  /**
+   *
+   * Theme based on which the text box is styled.
+   */
+  @Prop() state: 'normal' | 'warning' | 'error' = 'normal';
+
+  /**
+   * Default value displayed in the input box.
+   */
+  @Prop({ mutable: true }) value?: string | null = '';
+
+  // Events
+  /**
+   * Triggered when a value is selected or deselected from the list box options.
+   */
+  @Event() fwTelInput: EventEmitter;
+
+  /**
+   * Triggered when the list box comes into focus.
+   */
+  @Event() fwTelBlur: EventEmitter;
+
+  /**
+   * Triggered when clear icon is clicked.
+   */
+  @Event() fwTelInputClear: EventEmitter;
+
+  @Method()
+  async getLibMethods() {
+    return {
+      isValidPhoneNumber,
+      parsePhoneNumber(...args) {
+        return parsePhoneNumber.apply(this, args);
+      },
+    };
+  }
+
+  componentWillLoad() {
+    this.setPhoneNumberDetails(this.value);
+  }
+
+  private getSingleFormat(code = '', number = '') {
+    return '+' + code + number;
+  }
+
+  @Watch('value')
+  watchPhoneDetails(newValue, oldValue) {
+    if (newValue && oldValue && newValue !== oldValue) {
+      this.setPhoneNumberDetails(newValue);
+    }
+  }
+
+  setPhoneNumberDetails(input: string) {
+    if (input) {
+      try {
+        let userInput;
+        if (input.includes('+')) {
+          userInput = input;
+        } else {
+          userInput = '+' + input;
+        }
+        const details = parsePhoneNumber(userInput);
+        this.countryCode = details.country;
+        this.phoneCode = details.countryCallingCode;
+        this.phoneNumber = details.nationalNumber;
+        this.disablePhoneInput = false;
+        this.innerHintText = '';
+      } catch (error) {
+        this.disablePhoneInput = true;
+        this.phoneNumber = input;
+        this.innerHintText = messages[error.message] || '';
+      }
+    }
+  }
+
+  private onInputBlur = (event: Event) => {
+    const value = (event.target as HTMLInputElement).value;
+    this.phoneNumber = value;
+    const isValid = isValidPhoneNumber(value, this.countryCode as CountryCode);
+    this.fwTelBlur.emit({
+      event,
```

Comment:
We have an event naming standard. Usually events should be in the form,
{ event, name, value, meta }. Kindly update this accordingly in all events. 

Kindly refer to fw-checkbox component for reference.

Reply:
Fixed : ✅